lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -510,18 +511,84 @@ static bool reuseExistingInOut(Operation *op) {
   return true;
 }
 
+/// Return true if this is something that will get printed as a unary operator
+/// by the Verilog printer.
+/// TODO: This is copied from PrettifyVerilog. Create a common library for
+/// verilog expression stuffs.
+static bool isVerilogUnaryOperator(Operation *op) {
+  if (isa<comb::ParityOp>(op))
+    return true;
+
+  if (auto xorOp = dyn_cast<comb::XorOp>(op))
+    return xorOp.isBinaryNot();
+
+  if (auto icmpOp = dyn_cast<comb::ICmpOp>(op))
+    return icmpOp.isEqualAllOnes() || icmpOp.isNotEqualZero();
+
+  return false;
+}
+
+/// Return true if it is beneficial to spill the operation under the specified
+/// spilling heuristic.
+bool shouldSpillWire(Operation &op,
+                     LoweringOptions::WireSpillingHeuristic heuristic) {
+  assert(heuristic != LoweringOptions::SpillNone &&
+         "If none, we should not call `prettifyAfterLegalization` in the first "
+         "place");
+
+  // Simple expressions like unary operations or reads might be cloned by
+  // PrettifyVerilog, so don't spill wires to prevent name duplications.
+  if (isVerilogUnaryOperator(&op) || isa<ReadInOutOp>(op))
+    return false;
+
+  switch (heuristic) {
```

Comment:
Just noting that longer term, this will probably get a lot more complex. Starting here makes sense. But eventually we'll probably want to factor out functions or even small classes for handling each arm of the switch.

Reply:
Yeah, spitting these into small classes sounds good to me. Eventually each heuristic might become stateful, so probably inheritance & virtual functions would be right way to go in the longer term.